core/server/worker/src/main/java/alluxio/worker/netty/DataServerBlockReadHandler.java

```diff
@@ -83,62 +126,119 @@ public void close() throws IOException {
    * @param blockWorker the block worker
    * @param fileTransferType the file transfer type
    */
-  DataServerBlockReadHandler(ExecutorService executorService, BlockWorker blockWorker,
+  public DataServerBlockReadHandler(ExecutorService executorService, BlockWorker blockWorker,
       FileTransferType fileTransferType) {
     super(executorService);
     mWorker = blockWorker;
     mTransferType = fileTransferType;
   }
 
   @Override
-  protected boolean acceptMessage(Object object) {
-    if (!super.acceptMessage(object)) {
-      return false;
-    }
-    Protocol.ReadRequest request = ((RPCProtoMessage) object).getMessage().asReadRequest();
-    return request.getType() == Protocol.RequestType.ALLUXIO_BLOCK;
-  }
-
-  @Override
   protected void initializeRequest(Protocol.ReadRequest request) throws Exception {
     mRequest = new BlockReadRequestInternal(request);
   }
 
   @Override
-  protected DataBuffer getDataBuffer(Channel channel, long offset, int len) throws IOException {
-    LocalFileBlockReader blockReader =
-        (LocalFileBlockReader) ((BlockReadRequestInternal) mRequest).mBlockReader;
-    Preconditions.checkArgument(blockReader.getChannel() instanceof FileChannel,
-        "Only FileChannel is supported!");
-    switch (mTransferType) {
-      case MAPPED:
-        ByteBuf buf = channel.alloc().buffer(len, len);
+  protected DataBuffer getDataBuffer(Channel channel, long offset, int len) throws Exception {
+    openBlock(channel);
+    BlockReader blockReader = ((BlockReadRequestInternal) mRequest).mBlockReader;
+
+    if (mTransferType == TRANSFER && !(blockReader instanceof UnderFileSystemBlockReader)) {
```

Comment:
NIT: instead of `!(blockReader instanceof UnderFileSystemBlockReader)`, should we be more explicit here
`(blockReader instanceof LocalFileBlockReader)`?